src/ai/susi/server/api/aaa/UploadDataSettings.java

```diff
@@ -0,0 +1,101 @@
+package ai.susi.server.api.aaa;
+
+import ai.susi.DAO;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.server.*;
+import org.json.JSONObject;
+
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Created by dravit on 12/6/17.
+ * Servlet to upload file to data/settings folder
+ */
+public class UploadDataSettings extends AbstractAPIHandler implements APIHandler{
+    @Override
+    public String getAPIPath() {
+        return "/data/settings/upload";
```

Comment:
We should not have "phantasy api paths". All servlets have paths according to their class package. In this case the API path must be "/aaa/uploadDataSettings.json".